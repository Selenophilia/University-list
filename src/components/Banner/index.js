import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography, Button } from "@material-ui/core";
import Proptypes from "prop-types";
import image from "../../assets/images/2485381.png";

const useStyles = makeStyles((theme) => ({
  banner: {
    display: "flex",
    [theme.breakpoints.down("md")]: {
      flexDirection: "column"
    }
  },
  left: {
    width: "50%",
    padding: 100,
    [theme.breakpoints.down("md")]: {
      width: "100%",
      paddingLeft: "none",
      paddingRight: "none",
      textAlign: "center",
      alignItems: "space-between"
    }
  },
  right: {
    padding: 100,
    width: "50%",
    [theme.breakpoints.down("md")]: {
      display: "none"
    }
  },
  image: {
    height: "100%",
    width: "100%"
  },
  text: {
    marginTop: 20,
    marginBottom: 10,
    lineHeight: 1.5,
    fontSize: 56,
    fontWeight: "bold",
    textAlign: "left",
    [theme.breakpoints.down("md")]: {
      width: "100%",
      textAlign: "center"
    }
  },
  signin: {
    backgroundColor: "#3CB371",
    color: "#FFFFFF",
    width: 150,
    "&:hover": {
      backgroundColor: "#38a167"
    }
  }
}));

const Banner = ({ history }) => {
  const classes = useStyles();
  return (
    <div className={classes.banner}>
      <div className={classes.left} data-testid="left">
        <Typography variant="h2" className={classes.text}>
          Make your college application better instantly
        </Typography>
        <Button
          variant="contained"
          className={classes.signin}
          onClick={() => history.push("/explore")}
          data-testid="explore"
        >
          {"Explore"}
        </Button>
      </div>
      <div className={classes.right} data-testid="right">
        <img src={image} className={classes.image} />
      </div>
    </div>
  );
};

Banner.propTypes = {
  history: Proptypes.object.isRequired
};

export default Banner;
