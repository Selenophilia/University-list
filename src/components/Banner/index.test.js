import React from "react";
import { render, screen } from "@testing-library/react";
import Banner from ".";

const renderRoot = (props) => {
  return render(<Banner {...props} />);
};

it("renders Banner component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});

it("renders Banner left component correctly ", () => {
  renderRoot();
  expect(screen.getByTestId("left")).toBeInTheDocument();
});

it("renders Banner right component correctly ", () => {
  renderRoot();
  expect(screen.getByTestId("right")).toBeInTheDocument();
});
