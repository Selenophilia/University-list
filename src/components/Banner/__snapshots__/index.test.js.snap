// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Banner component correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-banner-1"
  >
    <div
      class="makeStyles-left-2"
      data-testid="left"
    >
      <h2
        class="MuiTypography-root makeStyles-text-5 MuiTypography-h2"
      >
        Make your college application better instantly
      </h2>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-signin-6"
        data-testid="explore"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Explore
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="makeStyles-right-3"
      data-testid="right"
    >
      <img
        class="makeStyles-image-4"
        src="2485381.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
